### PR TITLE
Question history plugin

### PR DESCRIPTION
**Pull request by safatshahin**

### CODE REVIEW

File: question/bank/history/templates/history_header.mustache [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          }
+        ]
+    }
+}}
+
+<div class="history-header mb-3">
+    <div class="row">
+        <div class="col-8 text-left font-weight-bold">
+            <h3>
+                {{{questionicon}}}
```
Comment (guillogo) [inline question/bank/history/templates/history_header.mustache:35]: Please add it to the doc above. Thank you.

File: question/bank/history/history.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+$returnurl = required_param('returnurl', PARAM_RAW);
+
+list($thispageurl, $contexts, $cmid, $cm, $module, $pagevars) =
+        question_edit_setup('questions', '/question/bank/history/history.php');
+
+$url = new moodle_url($thispageurl, ['entryid' => $entryid, 'returnurl' => $returnurl]);
+$PAGE->set_url($url);
+$questionbank = new \qbank_history\question_history_view($contexts, $url, $COURSE, $cm, $entryid, $returnurl);
+
+// Todo: make changes after bulk actions ui MDL-72076.
```
Comment (guillogo) [inline question/bank/history/history.php:39]: Please use the `// TODO MDL-12345` Thank you.

File: question/bank/history/classes/question_history_view.php [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    protected function display_question_bank_header(): void {
+        global $PAGE, $DB;
+        $sql = 'SELECT q.*
+                 FROM {question} q
+                 JOIN {question_versions} qv ON qv.questionid = q.id
+                 JOIN {question_bank_entry} qbe ON qbe.id = qv.questionbankentryid
+                WHERE qv.version  = (SELECT MAX(v.version)
+                                        FROM {question_versions} v
```
Comment (guillogo) [inline question/bank/history/classes/question_history_view.php:220]: Please align the query.

File: question/bank/history/history.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__ . '/../../../config.php');
+require_once($CFG->dirroot . '/question/editlib.php');
+require_login();
+$entryid = required_param('entryid', PARAM_INT);
+$returnurl = required_param('returnurl', PARAM_RAW);
+
```
Comment (guillogo) [inline question/bank/history/history.php:31]: please add the enable/disable plugin validation here: `\core_question\local\bank\helper::require_plugin_enabled(...`

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @safatshahin 

Thank you for working on this plugin, this is a very cool feature. I've finished the review just minors changes to do (added as a comments in the code).

Also, can you check these 2 things:

1. Should we need a behat test for the plugin?
2. Validate if history should be done as different tracker?

User test: Passed.

Gracias!

**Comment by safatshahin** [comment]
closing for branching reason, creating a new one.

**Pull request closed without merging**